docs: explain defaultRoute (#3917)
fastify/fastify: docs/Reference/Server.md

#### getDefaultRoute
<a id="getDefaultRoute"></a>

> The `defaultRoute` handler handles requests that do not match any URL specified by your Fastify application. This defaults to the 404 handler, but can be overridden with [setDefaultRoute](#setdefaultroute).

Reviewer: Please hard wrap lines to be consistent with the rest of the document.

Author: done

#### getDefaultRoute
<a id="getDefaultRoute"></a>

> `defaultRoute` is a handler that handles the requests that do not match any URL specified inside your application which defaults to 404 router but can be overridden with [setDefaultRoute](#setdefaultroute)
> 

Reviewer 2: ```suggestion
The `defaultRoute` handler handles requests that do not match any URL specified by your Fastify application. This defaults to the 404 handler, but can be overridden with [setDefaultRoute](#setdefaultroute).
```

#### setDefaultRoute
<a id="setDefaultRoute"></a>

> **Note**: The default 404 handler, or one set using `setNotFoundHandler`, will never trigger if the default route is overridden. Use [setNotFoundHandler](#setnotfoundhandler) if you want to customize 404 handling instead.

Reviewer: Please hard wrap lines to be consistent with the rest of the document.

Author: done

#### setDefaultRoute
<a id="setDefaultRoute"></a>

> **Important Note**: If you override the default route, it means that the 404 router will never be triggered. That also means `setNotFoundHandler` will not work, and no hooks fired. thus, if you want to customize handling 404s you can use `setNotFoundHandler` instead
> 

Reviewer 2: ```suggestion
**Note**: The default 404 handler, or one set using `setNotFoundHandler`, will never trigger if the default route is overridden. Use [setNotFoundHandler](#setnotfoundhandler) if you want to customize 404 handling instead.
```
